core/shared/src/main/scala/zio/Semaphore.scala

```diff
@@ -39,6 +39,12 @@ sealed trait Semaphore extends Serializable {
    */
   def available(implicit trace: Trace): UIO[Long]
 
+  /**
+   * Returns the number of tasks currently waiting for permits. The default
+   * implementation returns 0.
+   */
+  def awaiting(implicit trace: Trace): UIO[Long] = ZIO.succeed(0L)
```

Comment:
Any reason to have a default implementation? It feels error-prone. I think the implementation can just be consistent with `available` since they are quire similar.

Reply:
I think it's for done for bin-compat. Although I'm not sure how much of an actual problem it is since the trait is sealed

Reply:
In the current implementation I have removed default implementation, but that made MiMa unhappy. To make it happy I had to update MiMaSettings and add following line:
` exclude[ReversedMissingMethodProblem]("zio.Semaphore.awaiting")`

I am in favor of adding default implementation back to be binary compatible.

Absence of default implementation can lead to Runtime Errors such as `AbstractMethodError` or `NoSuchMethodError`. (if a user's application was not recompiled after the library upgrade)

Or

Even thought Semaphore trait is sealed, the user's code still might reference Semaphore in type signatures or interact with a Semaphore instances. (adding methods changes the interface's binary layout. The compiled code and the library must have matching binary interfaces to function correctly.)

These are my thoughts. Curious to hear what you think.